File: app/src/main/java/tech/pauly/findapet/shelters/SheltersFragment.java
package tech.pauly.findapet.shelters;

import android.content.Context;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import javax.inject.Inject;

import dagger.android.support.AndroidSupportInjection;
import tech.pauly.findapet.R;
import tech.pauly.findapet.databinding.FragmentSheltersBinding;

public class SheltersFragment extends Fragment {

    @Inject
    SheltersViewModel viewModel;

    @Override
    public void onAttach(Context context) {
        AndroidSupportInjection.inject(this);
        super.onAttach(context);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        FragmentSheltersBinding binding = DataBindingUtil.inflate(inflater, R.layout.fragment_shelters, container, false);
        getLifecycle().addObserver(viewModel);
        return binding.getRoot();
    }
}

File: app/src/main/java/tech/pauly/findapet/shelters/SheltersFragment.kt
package tech.pauly.findapet.shelters

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import tech.pauly.findapet.R
import tech.pauly.findapet.databinding.FragmentSheltersBinding
import tech.pauly.findapet.dependencyinjection.PetApplication
import tech.pauly.findapet.shared.BaseFragment
import tech.pauly.findapet.shared.MapWrapper
import javax.inject.Inject

class SheltersFragment : BaseFragment() {

    @Inject
    lateinit var viewModel: SheltersViewModel

    @Inject
    lateinit var mapWrapper: MapWrapper

    override fun onAttach(context: Context?) {
        PetApplication.component.inject(this)
        super.onAttach(context)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val binding = DataBindingUtil.inflate<FragmentSheltersBinding>(inflater, R.layout.fragment_shelters, container, false)
        addViewModelLifecycleObserver(viewModel)
        lifecycle.addObserver(mapWrapper)
        mapWrapper.setupMap(binding.mapView)
        binding.viewModel = viewModel
        return binding.root
    }
}
